### ISSUE TITLE
class_1921/cloth-client-events-v0.mixins.json crash on FO 2.x.x/3.x.x

### ISSUE THREAD

**Issue by Madis0** [Fabulously-Optimized/fabulously-optimized#192]
### Describe the bug

This cloth-client-events-v0.mixins.json error shows up on the first launch of FO
or
```
Error: java.lang.NoClassDefFoundError: Could not initialize class net.minecraft.class_1921
```

**Workaround: just launch again**

![](https://i.ibb.co/y0L1pBx/image.png)

### Expected behavior

No crash

### Observed/actual behavior

Crash

### Steps To Reproduce

1. Download modpack
2. Launch it

### Modpack version

3.1.0-alpha.1, 3.0.0-alpha.3, possibly others

### Launcher

Curseforge Launcher

### Install method

Fresh install/new profile

### Additional context

Second launch fixes it, and I'm not yet sure why this happens in the first place.
Emma suggested it may be due to duplicate Cloth API in the modpack (as Curseforge Launcher adds it automatically).
Edit: removing it didn't fix, but downgrading did... for now.

<details>

```
[18:06:22] [main/INFO]: Loading Minecraft 1.18.1 with Fabric Loader 0.12.11
[18:06:22] [main/INFO]: Fabric is preparing JARs on first launch, this may take a few seconds...
[18:06:29] [main/INFO]: Dependencies overridden for fabric, antighost, animatica, advancementinfo, dcch, colormatic
[18:06:29] [main/INFO]: Loading 119 mods:
	- advanced_runtime_resource_pack 0.5.4 via enhancedblockentities
	- advancementinfo 1.18-fabric0.43.1-1.3
	- animatica 0.1+1.17
	- antighost 1.18-fabric0.43.1-1.1.4
	- architectury 3.2.51
	- autoconfig1u 3.2.2 via capes
	- bettermounthud 1.1.3
	- borderlessmining 1.1.1+1.18
	- capes 1.2.2+1.18
	- cem 0.7.0
	- citresewn 0.8.1-1.18
	- cloth-api 2.0.54
	- cloth-basic-math 0.6.0 via cloth-config
	- cloth-client-events-v0 2.0.54 via cloth-api
	- cloth-common-events-v1 2.0.54 via cloth-api
	- cloth-config 6.1.48
	- cloth-datagen-api-v1 2.0.54 via cloth-api
	- cloth-scissors-api-v1 2.0.54 via cloth-api
	- cloth-utils-v1 2.0.54 via cloth-api
	- colormatic 2.4.0+mc.1.18
	- com_moandjiezana_toml_toml4j 0.7.2 via dynamicfps
	- completeconfig 1.3.3
	- continuity 1.0.3+1.18
	- crowdin-translate 1.4+1.18 via antighost
	- dcch 1.0
	- dynamicfps 2.0.6
	- enhancedblockentities 0.5+1.18
	- entityculling 1.3.3
	- fabric 0.44.0+1.18
	- fabric-api-base 0.4.1+b4f4f6cd14 via citresewn
	- fabric-api-lookup-api-v1 1.4.0+16d92c47c8 via fabric
	- fabric-biome-api-v1 6.0.1+ded849a9c8 via fabric
	- fabric-blockrenderlayer-v1 1.1.9+3ac43d95c8 via fabric
	- fabric-command-api-v1 1.1.6+3ac43d9514 via citresewn
	- fabric-commands-v0 0.2.5+b4f4f6cdc8 via fabric
	- fabric-containers-v0 0.1.18+d154e2c6c8 via fabric
	- fabric-content-registries-v0 0.4.5+6f53a73dc8 via fabric
	- fabric-crash-report-info-v1 0.1.8+3ac43d95c8 via fabric
	- fabric-dimensions-v1 2.1.7+43d29571c8 via fabric
	- fabric-entity-events-v1 1.4.5+6b21378ac8 via fabric
	- fabric-events-interaction-v0 0.4.16+bfa23f17c8 via fabric
	- fabric-events-lifecycle-v0 0.2.6+b4f4f6cdc8 via fabric
	- fabric-game-rule-api-v1 1.0.10+3ac43d95c8 via fabric
	- fabric-item-api-v1 1.3.0+691a79b5c8 via fabric
	- fabric-item-groups-v0 0.3.3+3ac43d95c8 via fabric
	- fabric-key-binding-api-v1 1.0.8+c8aba2f3c8 via fabric
	- fabric-keybindings-v0 0.2.6+b4f4f6cdc8 via fabric
	- fabric-language-kotlin 1.7.0+kotlin.1.6.0
	- fabric-lifecycle-events-v1 1.4.10+c15ca335c8 via fabric
	- fabric-loot-tables-v1 1.0.8+3ac43d95c8 via fabric
	- fabric-mining-level-api-v1 1.0.3+3ac43d95c8 via fabric
	- fabric-mining-levels-v0 0.1.7+b4f4f6cdc8 via fabric
	- fabric-models-v0 0.3.3+3ac43d95c8 via fabric
	- fabric-networking-api-v1 1.0.18+3ac43d95c8 via fabric
	- fabric-networking-v0 0.3.5+b4f4f6cdc8 via fabric
	- fabric-object-builder-api-v1 1.11.0+3b82842ec8 via fabric
	- fabric-object-builders-v0 0.7.8+3ac43d95c8 via fabric
	- fabric-particles-v1 0.2.9+526dc1acc8 via fabric
	- fabric-registry-sync-v0 0.8.5+3ac43d95c8 via fabric
	- fabric-renderer-api-v1 0.4.9+3ac43d95c8 via fabric
	- fabric-renderer-indigo 0.4.12+3ac43d95c8 via fabric
	- fabric-renderer-registries-v1 3.2.7+b4f4f6cdc8 via fabric
	- fabric-rendering-data-attachment-v1 0.3.4+7242e9d7c8 via fabric
	- fabric-rendering-fluids-v1 0.1.18+3ac43d95c8 via fabric
	- fabric-rendering-v0 1.1.9+b4f4f6cdc8 via fabric
	- fabric-rendering-v1 1.10.3+6b21378ac8 via fabric
	- fabric-resource-loader-v0 0.4.11+3ac43d9514 via borderlessmining
	- fabric-screen-api-v1 1.0.7+3ac43d95c8 via fabric
	- fabric-screen-handler-api-v1 1.1.11+3ac43d95c8 via fabric
	- fabric-structure-api-v1 2.0.8+295197a7c8 via fabric
	- fabric-tag-extensions-v0 1.2.5+3ac43d95c8 via fabric
	- fabric-textures-v0 1.0.9+3ac43d95c8 via fabric
	- fabric-tool-attribute-api-v1 1.3.4+7de09f55c8 via fabric
	- fabric-transfer-api-v1 1.5.5+b4f4f6cdc8 via fabric
	- fabricloader 0.12.11
	- fabrishot 1.6.0
	- fastopenlinksandfolders 1.0.0
	- ferritecore 4.0.0
	- indium 1.0.2-alpha1+mc1.18
	- iris 1.1.3
	- itemmodelfix 1.0.2+1.17
	- java 17
	- lambdabettergrass 1.2.1+1.17
	- lambdynlights 2.1.0+1.17
	- languagereload 1.2
	- lazydfu 0.1.2
	- lithium 0.7.6
	- minecraft 1.18.1
	- mm 2.1 via slight-gui-modifications
	- modmenu 3.0.0
	- morechathistory 1.1.0
	- no-telemetry 1.3.0
	- nofade 1.18-2.0.1
	- notenoughcrashes 4.1.3+1.18
	- org_anarres_jcpp 1.4.14 via iris
	- org_aperlambda_lambdajcommon 1.8.1 via spruceui
	- org_codehaus_groovy_groovy 3.0.8 via slight-gui-modifications
	- org_codehaus_groovy_groovy-jsr223 3.0.8 via slight-gui-modifications
	- org_jetbrains_kotlin_kotlin-reflect 1.6.0 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib 1.6.0 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk7 1.6.0 via fabric-language-kotlin
	- org_jetbrains_kotlin_kotlin-stdlib-jdk8 1.6.0 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm 1.5.2 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-coroutines-jdk8 1.5.2 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm 1.3.1 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-core-jvm 1.3.1 via fabric-language-kotlin
	- org_jetbrains_kotlinx_kotlinx-serialization-json-jvm 1.3.1 via fabric-language-kotlin
	- org_joml_joml 1.10.2 via sodium
	- org_slf4j_slf4j-api 1.7.12 via iris
	- pride 1.1.0+1.17 via lambdynlights
	- reeses-sodium-options 1.2.3
	- slight-gui-modifications 2.0.2
	- smoothboot 1.18-1.7.0
	- sodium 0.4.0-alpha5+build.9
	- sodium-extra 0.3.7
	- spruceui 3.3.0+1.17 via lambdabettergrass
	- tooltipfix 1.0.4-1.18
	- wi_zoom 1.3-MC1.18.1
	- yosbr 0.1.1
[18:06:34] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.4 Source=file:/C:/Users/Madis/curseforge/minecraft/Install/libraries/net/fabricmc/sponge-mixin/0.10.7+mixin.0.8.4/sponge-mixin-0.10.7+mixin.0.8.4.jar Service=Knot/Fabric Env=CLIENT
[18:06:34] [main/INFO]: Compatibility level set to JAVA_16
[18:06:34] [main/INFO]: Compatibility level set to JAVA_17
[18:06:34] [main/ERROR]: Uncaught exception in thread "main"
java.lang.RuntimeException: Error creating Mixin config cloth-client-events-v0.mixins.json for mod cloth-client-events-v0
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:90) ~[fabric-loader-0.12.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148) ~[fabric-loader-0.12.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:71) ~[fabric-loader-0.12.11.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.12.11.jar:?]
Caused by: org.spongepowered.asm.launch.MixinInitialisationError: Error initialising mixin config cloth-client-events-v0.mixins.json
	at org.spongepowered.asm.mixin.transformer.Config.create(Config.java:153) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.Mixins.createConfiguration(Mixins.java:100) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.Mixins.addConfiguration(Mixins.java:87) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:88) ~[fabric-loader-0.12.11.jar:?]
	... 3 more
Caused by: java.lang.IllegalArgumentException: The specified resource 'cloth-client-events-v0.mixins.json' was invalid or could not be read
	at org.spongepowered.asm.mixin.transformer.MixinConfig.create(MixinConfig.java:1290) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.transformer.Config.create(Config.java:148) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.Mixins.createConfiguration(Mixins.java:100) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at org.spongepowered.asm.mixin.Mixins.addConfiguration(Mixins.java:87) ~[sponge-mixin-0.10.7+mixin.0.8.4.jar:0.10.7+mixin.0.8.4]
	at net.fabricmc.loader.impl.launch.FabricMixinBootstrap.init(FabricMixinBootstrap.java:88) ~[fabric-loader-0.12.11.jar:?]
	... 3 more
```

</details>

**Comment by Hdhddufhrbtbtbfh848** [comment]
I was getting the same issue and it said "possible solution
Remove animators and dcch"

**Comment by Madis0** [comment]
That could be a potential solution, though I'm not that sure you got the same error. Which launcher did you install FO with?

**Comment by Hdhddufhrbtbtbfh848** [comment]
Vanilla

**Comment by Hdhddufhrbtbtbfh848** [comment]
New installition

**Comment by Madis0** [comment]
In that case you probably missed the `fabric_loader_dependencies.json` from `config`.

**Comment by Hdhddufhrbtbtbfh848** [comment]
> In that case you probably missed the `fabric_loader_dependencies.json` from `config`.

Ok

**Comment by Excaliburning** [comment]
Same issue.
![image](https://user-images.githubusercontent.com/9373373/146685306-1453cf49-92e8-4d12-8f90-ed99683eb7a9.png)

**Comment by Madis0** [comment]
Thank you, I am indeed aware that this is still an issue. 

Are you suggesting that overridden dependencies could cause it? That seems reasonable, but I'm not sure if I want to disable Animatica right now though. After all 3.1.0 is in alpha at the moment, so things like that are expected.

**Comment by Excaliburning** [comment]
> Are you suggesting that overridden dependencies could cause it?

Not at all, I provided the .json strings in case something was wrong in the provided file, if it seems alright with you then it is most likely not the cause. You've helped me plenty already in the comments section on CF.
Animatica works nicely so far, I see no reason to disable it either. Maybe a disclaimer that the first launch will crash and not to worry about it is enough for now, as I cannot understand the root of this issue myself (lack of knowledge of Fabric and Mixin). To be fair, it is a fairly minor one since it is fixed from the second launch and beyond.

**Comment by Madis0** [comment]
Disclamer is a good point, I forgot to do it so far :D

**Comment by Madis0** [comment]
Unpinned as it should no longer affect 3.x.x and it never did affect 1.x.x. 2.x.x remains an issue for the time being: https://github.com/shedaniel/cloth-api/issues/40

**Comment by nsde** [comment]
Is this issue still persistent?

**Comment by Madis0** [comment]
Unfortunately, I think yes.

**Comment by Madis0** [comment]
So, 4.0.0-alpha.4 was released which removed Cloth API entirely.

Since existent mods already worked in 4.0.0 alphas without it, it is probably safe to assume this dependency is no longer needed, persumably due to #18. Will close this issue when I'll update 1.18 and remove it from there as well.

**Comment by Madis0** [comment]
Fixed in 3.10.0.
